editor/src/components/canvas/controls/comment-indicator.tsx

```diff
@@ -538,24 +541,28 @@ function useHover() {
     setHovered(false)
   }, [])
 
-  return { hovered, onMouseOver, onMouseOut, cancelHover }
+  return { hovered, onMouseOver, onMouseOut, cancelHover, wasHovered }
 }
 
 type CommentIndicatorWrapper = {
   thread: ThreadData<ThreadMetadata>
   user: UserMeta | null
   expanded: boolean
+  animatable: boolean
   forceDarkMode: boolean
 } & CommentProps
 
 const CommentIndicatorWrapper = React.memo((props: CommentIndicatorWrapper) => {
-  const { thread, expanded, user, forceDarkMode, ...commentProps } = props
+  const { thread, expanded, user, forceDarkMode, animatable, ...commentProps } = props
 
   const [avatarRef, animateAvatar] = useAnimate()
 
   const animDuration = 0.1
 
   React.useEffect(() => {
+    if (avatarRef.current == null || !animatable) {
```

Comment:
when is this ref null? is this deterministic? is there a particular state / order of events that leads to a null ref?

Reply:
looking at the flow, it seems like it's when the component is mounted

Reply:
oh I agree too (and if you look at the code I always use the declarative approach), but there was definitely a reason for this `useAnimate`, which… I don't remember, and I looked for references in the code/past PRs… with no avail, so I just did that (and the bug is fixed too) [`954ad5a` (#5019)](https://github.com/concrete-utopia/utopia/pull/5019/commits/954ad5a892d6d69282a51a1016e7dd30bef3b86f)

happy monday folks!

Reply:
Win!

Reply:
woohoooo!